core/pyos.py

```diff
@@ -354,12 +376,16 @@ def AddUserTrap(self, sig_num, handler):
       self.sigwinch_handler.user_handler = handler
     else:
       signal.signal(sig_num, handler)
+    self.traps[sig_num] = handler
     # TODO: SIGINT is similar: set a flag, then optionally call user _TrapHandler
 
   def RemoveUserTrap(self, sig_num):
     # type: (int) -> None
     """For user-defined handlers registered with the 'trap' builtin."""
     # Restore default
+    if sig_num in self.traps:
```

Comment:
FWIW we have `mylib.dict_remove()`

https://github.com/oilshell/oil/blob/master/mycpp/mylib.py#L140

The benefit is that in C++ we won't involve any exceptions, which can be slow in the "main path" of the code

i.e. this is a shim, and in C++ there is another method that ensures key doesn't exist

Reply:
Ah nice I'll use that